Cython/Compiler/Nodes.py

```diff
@@ -8200,7 +8228,7 @@ def analyse_expressions(self, env):
         return self
 
     def generate_execution_code(self, code):
-        pass
+        cimport_numpy_check(self, code)
```

Comment:
I wonder if it's enough to add the helper function to [`utility_code_for_cimports`](https://github.com/cython/cython/blob/188ca7f9b873dc8b702a704da9ea83299a4e1f29/Cython/Compiler/Nodes.py#L8147-L8153) ?
You may have to pass `env` in there (to access `env.global_scope()`), but that should be ok, I think.

Reply:
`utility_code_for_cimports` is called during the AnalyseDeclarations phase, so we definitely haven't got `code`. We also haven't got the `entry.used` values set. I see the appeal of putting it there, but it doesn't really fit with how it currently works. It's not impossible though - would just need things moving around a bit.